client/verta/verta/pipeline/_registered_pipeline.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+
+import copy
+import json
+import tempfile
+from typing import Any, Dict, Optional
+
+from verta.endpoint.resources import Resources
+from verta.pipeline import PipelineGraph
+from verta.registry.entities import RegisteredModelVersion
+
+
+class RegisteredPipeline:
+    """Object representing a version of a registered inference pipeline.
+
+    There should not be a need to instantiate this class directly; please use
+    :meth:`Client.create_registered_pipeline() <verta.Client.create_registered_pipeline>`
+    for creating a new pipeline, or
+    :meth:`Client.get_registered_pipeline() <verta.Client.get_registered_pipeline>`
+    for fetching an existing pipeline.
+
+    .. note::
+        Registered pipelines are immutable once registered with Verta. A new version
+        must be created and registered with any desired changes. Use the ``copy_graph()``
+        function to create a local copy of this pipeline's graph that can be modified
+        and used to create the new version.
+
+    Attributes
+    ----------
+    name: str
+        Name of this pipeline.
+    id: int
+        ID of this Pipeline, auto-assigned by the Verta backend.
+    graph: :class:`~verta.pipeline.PipelineGraph`
+        PipelineGraph object containing all possible steps in the Pipline.
+    """
+
+    def __init__(
+        self,
+        registered_model_version: RegisteredModelVersion,
+        graph: PipelineGraph,
+    ):
+        """Create a Pipeline instance from an existing RegisteredModelVersion object
+        and the provided pipeline graph.
+
+        Name and ID are captured once upon creation to avoid additional HTTP calls
+        to refresh the cache of the RMV, because pipelines are immutable.
+        """
+        self._registered_model_version = registered_model_version
+        self._name = self._registered_model_version.name
+        self._id = self._registered_model_version.id
+        self._graph = graph
+        self._graph_steps = self._graph.steps
+        # throws an exception if the graph's steps attr has been inappropriately mutated.
+
+    def __repr__(self):
+        return "\n    ".join(
+            (
+                "RegisteredPipeline:",
+                f"pipeline name: {self.name}",
+                f"pipeline id: {self.id}",
+                f"\n{self._graph}",
```

Comment:
For consistency, since we're using the public attributes for name and ID

```suggestion
                f"\n{self.graph}",
```